ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -104,6 +106,14 @@ public AlterTableCompactOperation(DDLOperationContext context, AlterTableCompact
                 compactionRequest, ServerUtils.hostname(), txnHandler, context.getConf());
         parseCompactionResponse(compactionResponse, table, partitionMapEntry.getKey());
       }
+      // If Iceberg table had partition evolution, it will create compaction request without partition specification,
+      // and it will compact all files from old partition specs, besides compacting partitions of current spec in parallel.
+      if (DDLUtils.isIcebergTable(table) && table.getStorageHandler().isPartitioned(table) &&
```

Comment:
why check for iceberg?

Reply:
I added a comment in the code to explain why I added that.
```
// If Iceberg table had partition evolution, it will create compaction request without partition specification,
// and it will compact all files from old partition specs, besides compacting partitions of current spec in parallel.
```
It check for Iceberg because it is needed only for Iceberg.